c/include/nvtx3/nvtx3.hpp

```diff
@@ -611,6 +611,19 @@
 #define NVTX3_CONSTEXPR_IF_CPP14
 #endif
 
+// Macro wrappers for C++ attributes: [[maybe_unused]] and [[nodiscard]]
+#if defined(__has_cpp_attribute)
+  #if __has_cpp_attribute(maybe_unused)
+    #define NVTX3_MAYBE_UNUSED [[maybe_unused]]
+  #endif
+  #if __has_cpp_attribute(nodiscard)
+    #define NVTX3_NO_DISCARD [[nodiscard]]
+  #endif
+#else
```

Comment:
These need preprocessor checks for the C++ language version that implements these attributes, and we could also benefit from adding compiler-specific alternatives for broader support.